design/nfs.md

```diff
@@ -23,65 +23,107 @@ NFS supports multiple read/write simultaneously so a single share can be attache
 ## Design
 With this design Rook is exploring to providing another widely adopted storage option for admins and users of cloud-native environments. This deisign tends to automate NFS starting from its configuration (such as allowed hosts, read/write permissions etc.) to deployment and provisioning. The operations on NFS which cannot be done natively by Kubernetes will be automated.
 NFS doesn’t provide an internal provisioner for kubernetes, so Rook is needed as an external provisioner.
+This design uses NFS-Ganesha server and NFS v4.
 
 ### Initial Setup
 
 The flow of creating NFS backed storage in Rook is
 1. The settings are determined and saved in an NFS server CRD (rook-nfs.yaml)
 2. `kubectl create -f rook-nfs.yaml`
-3. When the NFS CRD instance is created, Rook responds to this request by starting the NFS daemon with the configuration and exports stated in the CRD and creates a service to expose NFS.
-   - Rook starts the NFS daemon on the desired nodes in the cluster
-   - NFS service is created
-   - Rook creates a storage class
+3. When the NFS CRD instance is created, Rook responds to this request by starting the NFS daemon with the required configuration and exports stated in the CRD and creates a service to expose NFS.
 4. NFS volume is ready to be consumed by other pods through a PVC.
 
 ### NFS CRD
 
 The NFS CRD spec will specify the following:
 1. NFS server storage backend configuration. E.g., configuration for various storage backends(ceph, ebs, azure disk etc) that will be shared using NFS.
-2. NFS server configuration
+2. NFS server configuration  
+   The following points are required for configuring NFS server:
     - export (The volume being exported)
     - client (The host or network to which the export is being shared)
     - client options (The options to be used for the client) e.g., read and write permission, root squash etc.
 
-The parameters to configure NFS are:
+The parameters to configure NFS CRD are:
 
 | Parameter                                 | Description                                                              | Default              |
 |-------------------------------------------|--------------------------------------------------------------------------|----------------------|
 | `replicas`                                | The no. of NFS daemon to start                                           | `1`                  |
-| `volumes`                                 | These are the volumes to be exposed by NFS server                        | <none>               |
+| `server`                                  | NFS server configuration                                                 | <none>               |
+| `server.volumeMount.name`                 | `volume.name` of the attached volume for sharing via NFS server          | <none>               |
```

Comment:
For each volume in the `volumes` element, there is a corresponding volumeMount in the `server` section, right? What if we combine these into a single list? For example:
```yaml
  volumes:
  - name: nfs-share
    mount:
       accessMode: ReadWrire
       squash: root
    persistentVolumeClaim:
      claimName: googlePD-claim
```

Reply:
Okay, and instead of `mount` we can name it `options` or `server`
```yaml
exports:
- name: nfs-share
  server:
      accessMode: ReadWrite
      squash: root
  persistantVolumeClaim:
    claimName: googlePD-claim
```
and for client specific example it can be
```yaml
exports:
- name: nfs-share
  server:
    allowedClients:
    - name: host1
      clients: 172.17.0.5
      accessMode: ReadOnly
      squash: root
    - name: host2
      clients:
      - 172.17.0.0/16
      - serverX
      accessMode: ReadWrire
      squash: none
  persistantVolumeClaim:
    claimName: googlePD-claim
```